### DISCUSSION TITLE
Configure OTLP within Milvus

### DISCUSSION THREAD

**Discussion by wolfganghuse** [milvus-io/milvus#28883] [Q&A and General discussion]
Hello, i deployed Milvus 2.3.3 with Helm chart on my K8s. I am running Open Telemetry together with Uptrace for Observability. Since Milvus 2.3 OTEL should be support but i didn´t find a good documentation.
By checking the source-code i understand that i can configure trace in milvus-config like:
```
trace:
  exporter: otlp
  otlp:
    endpoint: my-otlp.namespace:4317
```

However, i still don´t get any traces, etc... am i missing something ?

Regards,
Wolfgang

**Comment by LoveEachDay** [comment]
@wolfganghuse Could you try to set `trace.sampleFraction: 1` instead?

**Comment by wolfganghuse** [comment]
@LoveEachDay Thanks, that brings me one step further... but still struggeling:
I see these errors now with endpoint configured:
```
[2023/12/01 10:58:45.544 +00:00] [INFO] [tracer/tracer.go:71] ["Init tracer finished"] [Exporter=otlp]
[2023/12/01 10:58:45.545 +00:00] [WARN] [grpclog/grpclog.go:46] ["[core][Channel #77 SubChannel #78] grpc: addrConn.createTransport failed to connect to {\n  \"Addr\": \"localhost:4317\",\n  \"ServerName\": \"localhost:4317\",\n  \"Attributes\": null,\n  \"BalancerAttributes\": null,\n  \"Type\": 0,\n  \"Metadata\": null\n}. Err: connection error: desc = \"transport: authentication handshake failed: tls: first record does not look like a TLS handshake\""]
```

My OLTP Sidecar is not using grpc tls, but still i should be able to do insecure communication. Therefore i should be able to change endpoint to http://localhost:4317
According to [otlptracegrpc@v1.21.0](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.21.0) this is possible.
However, Milvus is still using v1.13.0 which does not support this, instead there is an additional .withinsecure function [.withInsecure@1.13.0](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.13.0#WithInsecure)

Can we update to newer otlptracegrpc, at least >=1.20, in that case we do not need to introduce additional Insecure-Parameter.

**Reply by LoveEachDay** [reply]
@wolfganghuse OK, we'll look into this issue shortly.
